indexd/drs/blueprint.py

```diff
@@ -0,0 +1,153 @@
+import flask
+
+from indexd.blueprint import dist_get_record
+
+from indexd.errors import AuthError
+from indexd.errors import UserError
+from indexd.index.errors import NoRecordFound as IndexNoRecordFound
+from indexd.drs.errors import UnexpectedError
+
+from indexd.fence_client import get_signed_url_for_object, FenceRequest
+
+try:
+    from local_settings import settings
+except ImportError:
+    from indexd.default_settings import settings
+
+import requests
+
+blueprint = flask.Blueprint("drs", __name__)
+
+blueprint.config = dict()
+blueprint.index_driver = None
+blueprint.dist = []
+
+@blueprint.route("/ga4gh/drs/v1/objects/<path:object_id>", methods=["GET"])
+def get_drs_object(object_id):
+    """
+    Returns a specific DRSobject with object_id
+    """
+    ret = blueprint.index_driver.get(object_id)
+    return flask.jsonify(indexd_to_drs(ret)), 200
+
+    
+@blueprint.route("/ga4gh/drs/v1/objects", methods=["GET"])
+def list_drs_records():
+    start = flask.request.args.get("page_token")
+    limit = flask.request.args.get("page_size")
+
+    try:
+        limit = 100 if limit is None else int(limit)
+    except ValueError:
+        raise UserError("limit must be an integer")
+
+    if limit <= 0 or limit > 1024:
+        raise UserError("limit must be between 1 and 1024")
+
+    url = flask.request.args.get("url")
+
+    checksum = flask.request.args.get("checksum")
+    if checksum:
+        hashes = {checksum["type"]: checksum["checksum"]}
+    else:
+        hashes = None
+
+    records = blueprint.index_driver.ids(
+        start=start, limit=limit, urls=url, hashes=hashes
+    )
+
+    ret = {"drs_objects": [indexd_to_drs(record, True)["drs_object"] for record in records]}
+
+    return flask.jsonify(ret), 200
+
+
+@blueprint.route("/ga4gh/drs/v1/objects/<path:object_id>/access/<path:access_id>", methods=["GET"])
+def get_signed_url(object_id, access_id=""):
+    res = get_signed_url_for_object(object_id, access_id)
+    if not res:
+        raise IndexNoRecordFound("No signed url found")
+    return res, 200
+
+
+def indexd_to_drs(record, list_drs = False):
+    bearer_token = flask.request.headers.get('AUTHORIZATION')
+    drs_object = {"id": record["did"], "description": "", "mime_type": ""}
```

Comment:
- file_name, created_date, updated_date, size, urls: these fields always exist in records so you don't need the `if` statements
- mime_type  should be `application/json`
- self_uri doesn't exist in indexd records, and the DRS spec says it's required. i think this current code would never include it, maybe @jiaqi216 can help figuring out what value to use
- contents and alias don't exist in indexd records, they are optional in the spec, they should probable always be included, but empty for now